app/admin/homepage.rb

```diff
@@ -0,0 +1,182 @@
+ActiveAdmin.register Homepage do
+  permit_params :internal_title, :section_title_one, :section_title_two, :section_title_three, :published, homepage_features_attributes: [:id, :title, :description, :url, :cta_text, :featured_image, :featured_image_file_name, :featured_image_content_type, :featured_image_updated_at, :featured_image_file_size, :image_alt_text, :delete_image, :section_id, :homepage_id, :position, :_destroy]
+
+
+  config.filters = false
+  config.sort_order = 'published_asc'
+
+  # Customizing the action items (buttons) on the show page
+  action_item :publish, priority: 0, only: :show do
+    publish_action_str = resource.published? ? 'Unpublish' : 'Publish'
+    link_to publish_action_str, publish_admin_homepage_path(resource), method: :post
+  end
+
+  member_action :publish, priority: 0, method: :post do
+    already_published = Homepage.where(published: true)
+    if resource.published
+      message = "\"#{resource.internal_title.to_s}\" unpublished"
```

Comment:
not super important but if there is no title for the homepage the publish / unpublish message appears with empy ""'s: 
![Screenshot 2024-08-29 at 11 39 16 AM](https://github.com/user-attachments/assets/021e5841-8b28-4be5-a5d9-c4258d1a7b14)